[Merged by Bors] - Fix incorrect rotation in `Transform::rotate_around`.

devil-ira opened the pull request:
Someone noted that the `rotate_around` method did not give the results they expected: [discord thread](https://discord.com/channels/691052431525675048/996497295325544479)
I tested `rotate_around` and their workaround and it seems like it was indeed incorrect.

Here is a scene with some cubes at different angles all being rotated around the center on the Y axis.

https://user-images.githubusercontent.com/29694403/178598432-407d7e80-1caf-4b17-b69b-66d9156c81e1.mp4

Interestingly, the middle cube rotates as you might expect. This threw me for a bit of a loop before I added the other cubes to the test haha.

Here is the same scene with the order multiplication of the quaternions flipped in `rotate_around`.

https://user-images.githubusercontent.com/29694403/178598446-a98026f3-524c-448b-8437-4d0d3175c6ca.mp4

That looks better :)

## Changelog

* Fixed `rotate_around` rotating the wrong way around
* Added `translate_around`. - Split out the translation code from `rotate_around`.
* Simplified/optimized `rotate_local_*` methods. - Yep, That works somehow.

<sup>Quaternions sure are wacky. Do not ask me how this works exactly, haha.</sup>

devil-ira commented:
I really need to stop forgetting to edit the PR title lol.
<sup> But I will never stop with the terrible commit messages hueheu /hj

IceSentry reviewed (approved):
The changes make sense to me. Should this be considered a breaking change since it would change the behavior of an api?

Nilirad commented:
> The changes make sense to me. Should this be considered a breaking change since it would change the behavior of an api?

I think not, because from what I understood, the current behavior is incorrect, therefore a bug.

devil-ira commented:
I added `rotate_local()` and `rotate_local_axis()` methods to mirror the existing non-local methods.
Also added to and edited the documentation a bit.

jpsikstus commented on crates/bevy_transform/src/components/transform.rs:
```diff
@@ -233,31 +235,53 @@ impl Transform {
+    /// The `rotation` is relative to this [`Transform`]'s current rotation.
+    #[inline]
+    pub fn rotate_local(&mut self, rotation: Quat) {
+        self.rotation *= rotation;
```
Does this not have the same issue of order of multiplication as what was fixed?

devil-ira replied:
This order of multiplication is not inherently wrong.
It's incorrect for `rotate_around` as that rotates around a point relative to the origin of the transform(ie. the parent if present, otherwise the origin of the world) not a local point relative to the transform itself.

alice-i-cecile commented:
bors r+
